### ISSUE TITLE
automatically disable/stop queue on instance restart after crash

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#4128]
Problem: if the rank 0 broker segfaults, systemd restarts it immediately.  The other ranks then fail and restart also, and the end result looks exactly like a healthy instance restart, except the KVS will have reverted to the last checkpiont, and the system loses track of any running jobs.

It seems like we need a way for the restarted rank 0 broker to detect that it was not shut down properly and prevent the system from starting back up again until a sys admin can intervene.

Related: #4092

**Comment by grondo** [comment]
> It seems like we need a way for the restarted rank 0 broker to detect that it was not shut down properly and prevent the system from starting back up again until a sys admin can intervene.

Are you thinking of something like a "safe mode" (where brokers can start but perhaps queue and some other features are disabled), or preventing the brokers from starting at all (which I think is what we had before a systemd config change)?

If the former, I wonder if there's a good way to notify users and sysadmins that the instance is operating in a safe mode?

**Comment by garlick** [comment]
I was thinking more of a safe mode at the instance level, where  scheduling and job submission would be disabled.

One idea would be to add an eventlog for instance restarts, where events are added for each instance start and stop.  If during startup, the last event posted was a start, safe mode could be entered.   This would need to be in conjunction with #4092 since otherwise the KVS would roll back to the previous clean shutdown.

**Comment by grondo** [comment]
> I wonder if we need another utility to provide at-a-glance status for a flux instance?

Since `broker.starttime` was added in #4076, I was thinking a `flux uptime` command might be a useful addition. In addition to the rank 0 "uptime" perhaps it could provide some other related instance status information?

**Comment by garlick** [comment]
I renamed this issue to focus it on a mechanism for preventing business as usual after an instance crash.   We can open another issue on `flux uptime` or other tooling.

**Comment by grondo** [comment]
Well I guess a trivial implementation would be an rc1 script that enables safe mode if a file exists in a predetermined location, creating the file but not activating safe mode if the file does not exist. 

A paired rc3 script would remove the file on normal shutdown.

This would also allow an administrator to setup safe mode before starting the broker (spiritual equivalent of `/etc/nologin`? The file could even contain an optional message).
